### ISSUE TITLE
[css-overflow-5] Bikeshed pseudoclasses for active scroll-marker

### ISSUE THREAD

**Issue by flackr** [w3c/csswg-drafts#10918]
Currently the [css-overflow-5 draft spec](https://drafts.csswg.org/css-overflow-5/#scroll-target) allows using `:checked` to style the active scroll marker.

See https://github.com/flackr/carousel/issues/28 for previous discussion on this.

On the semantics, a marker can be implicitly (through scrolling to that area) or explicitly (clicked on) activated. In either case, it remains active as long as the active marker calculation determines it to be so. As such, this should not be related to focus or direct interaction which go away when the user starts using other controls.  

Options:
1. Continue using `:checked`. The [HTML](https://html.spec.whatwg.org/multipage/semantics-other.html#selector-checked) and [css-selectors](https://drafts.csswg.org/selectors/#checked) spec say that checked represents a selected-ness. Scroll markers can be selected by the user explicitly or implicitly by scrolling, and cannot be independently checked as they cannot be radio buttons, checkboxes or select options so the use cases when checkedness applies shouldn't overlap.
2.  Use `:current` as suggested in https://github.com/flackr/carousel/issues/28#issue-2406219421. As @tabatkins points out, this already has a meaning independent of markers which means both sets of meanings could apply making this potentially confusing.
3. Use `:active`. This has the right sound to it, but `:active` already has a meaning which should separately apply to a scroll-marker when it is being interacted with.
4. Something else?

**Comment by fantasai** [comment]
New proposal from me and @flackr: `:target-shown`

**Comment by Schepp** [comment]
Hey there, first and foremost, it's wonderful to see this being addressed. It might be bike shedding, but it has the potential to set the path forward for many upcoming specs, too.

I did a quick check by issuing `document.querySelector(':current')` and `document.querySelector(':current(p)')` commands in their developer consoles and apparently none of the three major rendering engines seemed to have implemented the `:current` selector so far. This means, it can still be repurposed to fit broader needs.

That said, I do like `:target-shown` a lot. It would also work with the Carousel API.

It also solves the problem that while `:current` might be familiar as it reminds people of `aria-current`, it would not work exactly the same way and confuse people. I find `aria-current` to exhibit more of a static nature: It marks the current document as the selected one in a navigation tree, as the current step in a multi step wizard or as the current date in a calendar. The CSS pseudo class on the other hand would be a lot more fluid and it would change multiple times during a page's lifecycle.

**Comment by kbabbitt** [comment]
> New proposal from me and @flackr: `:target-shown`

I prefer this over any of the options we discussed yesterday, though I could still live with `:current` if we don't get traction on `:target-shown` or some other alternative.

My reservation about `:current` is less about the not-yet-implemented WebVTT usage and more about the word "current" being ambiguous when read on its own. "Current what?" is the question I'm left with. See also `currentcolor` which answers that question and `aria-current` which doesn't (and I've had to go look it up to remind myself in the past).

**Comment by johannesodland** [comment]
As I read the spec (and I might have misunderstood), only one scroll marker control can have the `checked` status, while there might be more than one scrollTargetElement visible in the scrollport. 

I’d expect `:target-shown` to match every control whose scrollTargetElement is visible in the scroll port, and not just one. Maybe because it sounds a bit like `:target-visible`?

**Comment by flackr** [comment]
> I’d expect :target-shown to match every control whose scrollTargetElement is visible in the scroll port, and not just one. Maybe because it sounds a bit like :target-visible?

Right, I think it's a good thing to avoid visible. Whichever name we go with we would of course have clear text explaining the meaning.

The relation to [:target](https://drafts.csswg.org/selectors/#the-target-pseudo) is nice, though the behavior is not identical - as this will change with scrolling whereas `:target` would only change if the URL was updated. Some other options:
1. `:scroll-target`: perhaps a bit odd to have this match on the marker pointing to the target, rather than the target itself.
2. `:selected`: easily confused with text selection which uses `::selection`, also could be confused with `<select>` which uses `:checked` which is what we're trying to move away from.
4. `:current-target`: answers the "current what?" question from @kbabbitt and might be sufficiently distinct from both `:current` and `:target`
5. `:target-current`: while it doesn't conflict with the usage of `:current`, it could still be conflated for being related to `:current`.
6. `:target-nearest` (or some variation on "nearest"): feels a bit awkward but does match pretty closely to what is meant
7. `:target-active`: Doesn't actually mean that the target is `:active`, but maybe?

> it has the potential to set the path forward for many upcoming specs, too.

No pressure 😁 

This also suggests we should think about what terminology might make sense for similar non-form non-focus related active / selected states, e.g. the current tab

**Comment by flackr** [comment]
After thinking through the options, I propose `:target-current` as the matching pseudo-class on the active scroll-marker.
* This follows precedent of the adjective following the related known used by `:focus-within`, `:focus-visible`, `:popover-open`.
* This relates to `:target` in that the `:target` will initially have its marker be `:target-current`.
* `:target-shown` suggests a relation to visibility, whereas other markers can have visible targets but not be active.

**Comment by fantasai** [comment]
Clearly the correct answer is `:target-targetted`. :)

**Comment by tabatkins** [comment]
Going with :target-current in the draft for now, just to make it stop suggesting that we're reusing :checked. We can edit it when we decide on a name for real. ^_^

**Comment by flackr** [comment]
Removing Needs Edits, as `:target-current` is what's reflected in the draft spec per https://github.com/w3c/csswg-drafts/issues/10918#issuecomment-2486772063
